### ISSUE TITLE
Classification with Discriminator 

### ISSUE THREAD

**Issue by hashbangCoder** [jsyoon0823/TimeGAN#11]
Hello, Thanks for providing the code for your paper

Could you please explain why the discriminator's (who's function is to classify each sample as real/fake) output [Y_fake](https://github.com/jsyoon0823/TimeGAN/blob/master/timegan.py#L177) (for example) is a 3D tensor?

More specifically, shouldnt the [classification CE loss](https://github.com/jsyoon0823/TimeGAN/blob/master/timegan.py#L190) - be classifying each sequence (aka sample ) instead of classifying at the time-step level?

Would this change make sense in this regard?

```
def discriminator (H, T):    
    with tf.variable_scope("discriminator", reuse = tf.AUTO_REUSE):
      d_cell = tf.nn.rnn_cell.MultiRNNCell([rnn_cell(module_name, hidden_dim) for _ in range(num_layers)])
     # use output for last time step instead
      d_outputs, d_last_states = tf.nn.dynamic_rnn(d_cell, H, dtype=tf.float32, sequence_length = T)
      Y_hat = tf.contrib.layers.fully_connected(d_outputs[:, -1, :], 1, activation_fn=None) 
    return Y_hat  
```

**Comment by jsyoon0823** [comment]
Thank you for your interest in our paper.
That is a good point. 
What we thought is that if we provide timely feedback to the generator, the generator can utilize more fruitful feedback.
Like whether the time-series (until t) seems real or fake.
So, we have not tried 2D discriminator output (like what you said that time-series itself is real/fake).
It seems like it would be worth to try.
